src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -36,28 +36,44 @@ class AppTasksResource @Inject() (service: MarathonSchedulerService,
 
   @GET
   @Timed
-  def indexJson(@PathParam("appId") appId: String,
+  def indexJson(@PathParam("appId") id: String,
                 @Context req: HttpServletRequest, @Context resp: HttpServletResponse): Response = {
-    doIfAuthorized(req, resp, ViewAppOrGroup, appId.toRootPath) { implicit principal =>
+    doIfAuthenticated(req) { identity =>
       val taskMap = taskTracker.tasksByAppSync
 
-      def tasks(appIds: Set[PathId]): Set[EnrichedTask] = for {
+      def runningTasks(appIds: Set[PathId]): Set[EnrichedTask] = for {
+        runningApps <- appIds.filter(taskMap.hasAppTasks)
         id <- appIds
         health = result(healthCheckManager.statuses(id))
         task <- taskMap.appTasks(id)
       } yield EnrichedTask(id, task, health.getOrElse(task.getId, Nil))
 
-      val matchingApps = appId match {
+      id match {
         case GroupTasks(gid) =>
-          result(groupManager.group(gid.toRootPath))
-            .map(_.transitiveApps.map(_.id))
-            .getOrElse(Set.empty)
-        case _ => Set(appId.toRootPath)
+          val groupPath = gid.toRootPath
+          result(groupManager.group(groupPath)) match {
+            case Some(group) =>
+              doIfAuthorized(req, identity, ViewGroup, group) { identity =>
+                val tasks = runningTasks(group.transitiveApps.map(_.id))
+                // FIXME (gkleiman): if tasks is empty, shouldn't we just return an empty tasks list?
+                if (tasks.isEmpty) unknownGroup(groupPath)
+                else ok(jsonObjString("tasks" -> tasks))
+              }
+            case None => unknownGroup(groupPath)
+          }
+        case _ =>
+          val appId = id.toRootPath
+          service.getApp(appId) match {
+            case Some(app) =>
+              doIfAuthorized(req, identity, ViewApp, app) { identity =>
+                val tasks = runningTasks(Set(appId))
+                // FIXME (gkleiman): if tasks is empty, shouldn't we just return an empty tasks list?
```

Comment:
It didn't, see https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala#L61 :+1: 

```scala
val running = matchingApps.filter(taskMap.hasAppTasks)

if (running.isEmpty) unknownApp(appId.toRootPath) else ok(jsonObjString("tasks" -> tasks(running)))
```